src/main/java/com/epam/javacourse/hometask10/base/dao/GreenhouseDao.java

```diff
@@ -0,0 +1,214 @@
+    @Override
+    public Greenhouse get(int id) {
+        String sql = "SELECT * FROM greenhouses WHERE id=?;";
+        try (PreparedStatement statement = ConnectionManager.getConnection().prepareStatement(sql)) {
+            statement.setInt(1, id);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                return new Greenhouse(id,
+                        resultSet.getString("name").trim(),
+                        resultSet.getInt("gardener_id"));
+            } else {
+                System.out.println("No result found for ID: " + id);
+                return null;
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new RuntimeException();
+        } finally {
+            ConnectionManager.closeConnection();
+        }
+    }
+
+    @Override
+    public List<Greenhouse> getAll() {
+        String sql = "SELECT * FROM greenhouses;";
+        try (Statement statement = ConnectionManager.getConnection().createStatement()) {
+            ResultSet resultSet = statement.executeQuery(sql);
+            List<Greenhouse> resultList = new ArrayList<>();
+            while (resultSet.next()) {
+                resultList.add(new Greenhouse(resultSet.getInt("id"),
+                        resultSet.getString("name"),
+                        resultSet.getInt("gardener_id")));
+            }
+            return resultList;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new RuntimeException();
+        } finally {
+            ConnectionManager.closeConnection();
+        }
+    }
+
+    @Override
+    public List<Greenhouse> getBetween(String column, int start, int end) {
+        String sql = "SELECT * FROM greenhouses WHERE ? BETWEEN ? AND ?;";
+        try (PreparedStatement statement = ConnectionManager.getConnection().prepareStatement(sql)) {
+            statement.setString(1, column);
+            statement.setInt(2, start);
+            statement.setInt(3, end);
+            ResultSet resultSet = statement.executeQuery();
+            List<Greenhouse> resultList = new ArrayList<>();
+            while (resultSet.next()) {
+                resultList.add(new Greenhouse(resultSet.getInt("id"),
+                        resultSet.getString("name").trim(),
+                        resultSet.getInt("gardener_id")));
+            }
+            return resultList;
+        } catch (SQLException e) {
+            e.printStackTrace();
+            throw new RuntimeException();
+        } finally {
+            ConnectionManager.closeConnection();
+        }
+    }
+
+    @Override
+    public List<Greenhouse> getIn(String column, List<String> list) {
+        StringJoiner joiner = new StringJoiner(",", "(", ")");
+        for (String o : list) {
+            joiner.add(o);
+        }
+        String sql = "SELECT * FROM greenhouses WHERE ? IN " + joiner + ";";
```

Comment:
It might worth moving joiner to a separate method in order not to duplicate code.